pkg/scaling/resolver/aws_secretmanager_handler.go

```diff
@@ -0,0 +1,120 @@
+
+	if ash.secretManager.Cloud != nil {
+		if ash.secretManager.Cloud.Region != "" {
+			config.Region = ash.secretManager.Cloud.Region
+		}
+	}
+
+	ash.session = secretsmanager.NewFromConfig(*config)
+	return nil
+}
+
+func (ash *AwsSecretManagerHandler) getcredentials(ctx context.Context, client client.Client, logger logr.Logger, triggerNamespace string, secretsLister corev1listers.SecretLister, podTemplateSpec *corev1.PodTemplateSpec) (*aws.Config, error) {
+	config := &aws.Config{}
+
+	podIdentity := ash.secretManager.PodIdentity
+	if podIdentity == nil {
+		podIdentity = &kedav1alpha1.AuthPodIdentity{}
+	}
+
+	switch podIdentity.Provider {
+	case "", kedav1alpha1.PodIdentityProviderNone:
+		accessKeyID := resolveAuthSecret(ctx, client, logger, ash.secretManager.Credentials.AccessKey.ValueFrom.SecretKeyRef.Name, triggerNamespace, ash.secretManager.Credentials.AccessKey.ValueFrom.SecretKeyRef.Key, secretsLister)
+		accessSecretKey := resolveAuthSecret(ctx, client, logger, ash.secretManager.Credentials.AccessSecretKey.ValueFrom.SecretKeyRef.Name, triggerNamespace, ash.secretManager.Credentials.AccessSecretKey.ValueFrom.SecretKeyRef.Key, secretsLister)
+		if accessKeyID == "" || accessSecretKey == "" {
+			return nil, fmt.Errorf("AccessKeyID and AccessSecretKey are expected when not using a pod identity provider")
+		}
+		config.Credentials = credentials.NewStaticCredentialsProvider(
+			accessKeyID,
+			accessSecretKey,
+			"",
+		)
+		return config, nil
+
+	case kedav1alpha1.PodIdentityProviderAwsEKS:
+		awsRoleArn, err := ash.getRoleArnAwsEKS(ctx, client, logger, triggerNamespace, podTemplateSpec)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving role arn for AwsEKS pod identity: %w", err)
+		}
+		config.Credentials = stscreds.NewAssumeRoleProvider(sts.NewFromConfig(*config), awsRoleArn)
+		return config, nil
+	case kedav1alpha1.PodIdentityProviderAwsEKSWorkload:
+		awsRoleArn, err := ash.getRoleArnAwsEKS(ctx, client, logger, triggerNamespace, podTemplateSpec)
+		if err != nil {
+			return nil, fmt.Errorf("error resolving role arn for AwsEKS workload identity: %w", err)
+		}
+		config.Credentials = stscreds.NewAssumeRoleProvider(sts.NewFromConfig(*config), awsRoleArn)
+		return config, nil
+	default:
+		return nil, fmt.Errorf("pod identity provider %s not supported", podIdentity.Provider)
+	}
+}
+
+func (ash *AwsSecretManagerHandler) getRoleArnAwsEKS(ctx context.Context, client client.Client, logger logr.Logger, triggerNamespace string, podTemplateSpec *corev1.PodTemplateSpec) (string, error) {
+	serviceAccountName := podTemplateSpec.Spec.ServiceAccountName
+	serviceAccount := &corev1.ServiceAccount{}
+	logger.Info("serviceAccount value", "serviceAccount", serviceAccount)
+	err := client.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: triggerNamespace}, serviceAccount)
+	if err != nil {
+		return "", err
+	}
+
+	return serviceAccount.Annotations[kedav1alpha1.PodIdentityAnnotationEKS], nil
```

Comment:
We still need to check if we are using operator identity or workload, because some users can prefer to grant permissions to KEDA role instead of always assuming other roles

Reply:
I need more clarity on this.
could you please clarify more?

Reply:
Using this code, the roleArn used is always the scaled workload roleArn, but KEDA can has it's own role. Indeed, probably for this, using KEDA role in the beginning looks like a better option that using the scaled workload role